sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -530,8 +531,15 @@ def post_port_sfp_firmware_info_to_db(logical_port_name, port_mapping, table,
             continue
 
         try:
-            transceiver_firmware_info_dict = _wrapper_get_transceiver_firmware_info(physical_port)
-            if transceiver_firmware_info_dict is not None:
+            if firmware_info_cache is not None and physical_port in firmware_info_cache:
```

Comment:
@mihirpat1 what is this optimization? My understanding is the dom is going to read the firmware version from the EEPROM every polling interval

Reply:
@prgeor - Upon further looking into the code after our discussion, I found that the current code reads data from EERPOM for the first subport for every poll. However, the remaining subports of that physical port in that iteration have their data retrieved from the relevant cache.
In the current case, the firmware_info_cache is a dictionary with `physical_port` as the key. Hence, DomInfoUpdateTask will clear the entire firmware_info_cache dictionary (line 1642) before it starts retrieving data for all ports. While handling the first subport of a physical port, DomInfoUpdateTask will read data from EEPROM and update firmware_info_cache. All other subports of the physical ports will have the data retrieved from firmware_info_cache in that iteration.